### DISCUSSION TITLE
Connect admin, medusa-store, nextjs-starter, Minio. Structure guidelines needed

### DISCUSSION THREAD

**Discussion by speciale** [medusajs/medusa#2848] [Q&A] [answered]
### Short version
I need some guidelines to build this monster. It's a complex architecture and I'm a newbie. I believe that if I can have a Postgres database and some images to start the store I could do it. But I'm a newbie with Node servers. It's my first attempt. I say this so y'all don't waste your time thinking I'm Tim Berners Lee like.

### Full (mostly) version
I'm coming from Discord. I'm trying to build a server for a friend. He's an excellent cook and it's going to start a "traiteur" service. I know pretty well CSS, HTML, PHP, HTTPD… I'm a graphic designer at start.

I need some guidelines. I did install 3 medusa gits following the docs. My first surprise is the difference with apache, where sites are in a folder which HTTPD uses to expose them. With Node you can spawn a server everywhere. But I have concerns with the structure.  I have this (like the img attached)

medu  > admin
           > my-medusa
           >  nextjs-starter-medusa

In some cases I use medusa develop, in others npm, in others yarn (blindly following the instructions). And in some cases I overlapped yarn and npm. Was all you can imagine but nice. But I still breathing. 
The result is that I get some results but they are like the walking dead monsters. No brains. So they can't talk to each other. 

### Problem at login

I can't login. 
```
npm run start
  medusa-admin@0.1.0 start
  npm run dev
  medusa-admin@0.1.0 dev
  vite --port 7000
  VITE v3.2.5  ready in 686 ms
  ➜  Local:   http://localhost:7000/
  ➜  Network: use --host to expose
```
It says: "Welcome back! It's great to see you" But then… "These credentials do not match our records". 
I did try admin@medusa-test.com supersecret. No luck with medusa user -e myname@gmail.com -p mypassword.

my .env is

```
JWT_SECRET=UneLonguePrahse
COOKIE_SECRET=EncoreUneLonguePhrase
STRIPE_API_KEY=pk_test_blablabla
STRIPE_WEBHOOK_SECRET=sk_test_blablabla
MINIO_ENDPOINT=http://127.0.0.1:9091
MINIO_BUCKET=my-medusa
MINIO_ACCESS_KEY=LeAccesDeMinio
MINIO_SECRET_KEY=LeSecretDeMinio
```
my medusa-config.js is (at he end)

```
const plugins = [
  `medusa-fulfillment-manual`,
  `medusa-payment-manual`,
  // Uncomment to add Stripe support.
  // You can create a Stripe account via: https://stripe.com
  {
    resolve: `medusa-payment-stripe`,
    options: {
      api_key: STRIPE_API_KEY,
      webhook_secret: STRIPE_WEBHOOK_SECRET,
    },
  },
  {
    resolve: `medusa-file-minio`,
    options: {
        endpoint: process.env.MINIO_ENDPOINT,
        bucket: process.env.MINIO_BUCKET,
        access_key_id: process.env.MINIO_ACCESS_KEY,
        secret_access_key: process.env.MINIO_SECRET_KEY,
    },
},
];
module.exports = {
  projectConfig: {
    redis_url: REDIS_URL,
    // For more production-like environment install PostgresQL
    // database_url: DATABASE_URL,
    //database_type: "postgres",
    database_database: "./medusa-db.sql",
    database_type: "sqlite",
    store_cors: STORE_CORS,
    admin_cors: ADMIN_CORS,
  },
  plugins,
};
```
### Problem at front

```
yarn dev
yarn run v1.22.19
$ next dev -p 8000
ready - started server on 0.0.0.0:8000, url: http://localhost:8000
info  - Loaded env from /Users/speciale/medu/nextjs-starter-medusa/.env.local
next.config.js {
  "reactStrictMode": true,
  "images": {
    "domains": [
      "medusa-public-images.s3.eu-west-1.amazonaws.com",
      "localhost"
    ]
  },
  "env": {
    "FEATURE_SEARCH_ENABLED": true
  }
}
```
Then a beautiful young woman says that…

```
   const createNewCart = async (regionId?: string) => {
     await createCart.mutateAsync(
                    ^
       { region_id: regionId },
       {
         onSuccess: ({ cart }) => {

_callee$
src/lib/context/store-context.tsx (219:14)
StoreProvider/<
src/lib/context/store-context.tsx (224:6)
```
And…
`Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at http://localhost:9000/store/regions. `
If I well understood that's a Minio problem. But I did what I found in the docs (bucket + keys). And Minio is working. Redis is working. Meilisearch too (I did not create the key yet). 
Thanks, if you got till here, for your empathy.

**Comment by pKorsholm** [comment, marked as answer]
Hey man!

wrt. npm/yarn etc you should be good just running yarn everywhere 😄 

it looks like Minio is not your problem atm, rather it looks like a mis-config of you CORS variables. Have you changed the `ADMIN_CORS` and `STORE_CORS` of the default `medusa-config.js` from the installation?

**Reply by ghost** [reply]
Which file should the `ADMIN_CORS=` and the `STORE_CORS=`live in - I might have moved them myself, but dont remember where I moved them from

**Comment by speciale** [comment]
Hi! My problem is that I'm a mono-neuronal  😈. First thanks for answering. Then, the more I work whit this, the more I see how stupid my post was. So I did a complete reinstall several times and now I'm beginning to "see the light". I will try to find what or where the problem is and then I'll come back to ask more questions if needed. Occurs that I have skip steps with my installs and now I'm trying to work by the book. I'm setting Redis actually. Again, thank you for your kindness. Warms the hearth I have not. 😈

**Comment by pKorsholm** [comment]
Alright, feel free to ask if you run in to any issue 😄

**Comment by speciale** [comment]
News from the front. I think that we have a problem, my future dear friends. It doesn't works… with Firefox. But it works with… Chrome. 
So far with Firefox it tells me…

<img width="707" alt="Screenshot 2022-12-22 at 19 28 11" src="https://user-images.githubusercontent.com/1444636/209203472-aabd23b4-334f-415d-a479-fae0da50c0f3.png">

But with Chrome I can login

<img width="825" alt="Screenshot 2022-12-22 at 19 28 44" src="https://user-images.githubusercontent.com/1444636/209203613-9cf82dce-28ba-4dbd-975c-17dfde9c4c6a.png">

I'm still having some issues

<img width="851" alt="Screenshot 2022-12-22 at 19 30 11" src="https://user-images.githubusercontent.com/1444636/209203752-9af707d2-5494-4031-8069-a74bc63b659d.png">

But I believe that I'm now far from the perspective of having to suicide myself with a feather.

The question now, if somebody feels that an answer is to consider, is how can it be possible?
What an I missing?

Thanks in advance. Il will keep you posted if this monster doesn't kill me before I found why this happens 😈

**Comment by speciale** [comment]
Still talking to myself. The problem I had wasn't Medusa or Firefox. The problem was that I had PHP-FPM running at… port 9000, which is the port he uses by default. Sorry (and ashamed) for the noise. Now I can begin to work with the others issues, having a relatively good start point.

**Comment by speciale** [comment]
I'm beginning now the customization and I can't see the modifications I made in the pages reflected in the browser. It's the nextjs-starter-medusa who appears to have a cache. 
So my question is how can I solve this? I mean, see in the browser the modifications I make?

**Reply by speciale** [reply]
I found it ! Instead of running yarn run start, I do now yarn run dev. Is this the appropriate way? So far it works  😈

**Comment by speciale** [comment]
Well, I thing that I have now a good starting point. In any case, if someone has PHP-FPM running on port 9000 they must change this port. Interesting point is that Chrome accepts this, but Firefox blocks interaction. 
Too many, several things I learned. Thanks and sorry for the noise.
